src/dcp/dcp.c

```diff
@@ -396,35 +402,38 @@ int main(int argc, char** argv)
     }
 
     /* setgroups before set gid or uid */
-    if (gid > 0 || uid > 0) {
+    if (egid > 0 || euid > 0) {
         if (setgroups(0, NULL) < 0) {
             MFU_LOG(MFU_LOG_ERR, "Could not setgroups: %s", strerror(errno));
             mfu_finalize();
             MPI_Finalize();
             return 1;
         }
+
+        /* record the original groups */
+        getgroups(MAX_GIDS, &gids);
```

Comment:
Please check this for failure and report it (e.g. errno==EINVAL means MAX_GIDS was too small), or dynamically get the size needed and allocate the buffer for the gids.